x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -579,6 +579,7 @@ export function removeLayer(layerId) {
       type: REMOVE_LAYER,
       id: layerId
     });
+    return Promise.resolve();
```

Comment:
What? Why does this return Promise.resolve? There is nothing that makes this method async.

Reply:
Rolled back to be addressed separately. We needs a means of confirming layer removal before subsequently adding a layer. We're only really in this situation when using file upload as we add one type of layer (geojson), remove it, then add another type of layer (ES Source) on index